deploy/balena-k3s/server/Dockerfile

```diff
@@ -1,13 +1,45 @@
-# https://hub.docker.com/r/rancher/k3s/tags
+# References:
+# https://k3d.io/v5.7.4/usage/advanced/cuda/#building-a-customized-k3s-image
 # https://github.com/k3s-io/k3s/blob/master/package/Dockerfile
-# https://github.com/balena-io-experimental/balena-k3s/blob/main/server/Dockerfile
-FROM rancher/k3s:v1.26.14-k3s1
 
-COPY server.sh /server.sh
-RUN chmod +x /server.sh
+ARG K3S_TAG="v1.28.8+k3s1"
+
+# Use this instead of, for example, balenalib/genericx86-64-ext-ubuntu:focal, because
+# it is cross-platform and can be used on any architecture / type of device.
+FROM ubuntu:22.04
+
+ENV DEBIAN_FRONTEND noninteractive
+
+# Install Nvidia driver. Must match version in gpu container exactly
+RUN \
+    apt-get update && \
+    apt install -y curl runc software-properties-common && \
+    add-apt-repository ppa:graphics-drivers/ppa && \
+    apt update && \
+    apt install -y nvidia-driver-550 \
```

Comment:
wrt the other problem we had: does this pull in `nvidia-smi` which the container toolkit depends on?

Reply:
Yes, this is what is pulling in `nvidia-smi` here